### PR TITLE
[stable/nginx-ingress] Remove obsolete stats configuration (#16137)

### LINKED ISSUE #16137: [nginx-ingress] Remove obsolete stats configuration
**A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]**
nginx-ingress does no longer expose a nginx server for stats at port 18080, see [Changelog.md](https://github.com/kubernetes/ingress-nginx/blob/master/Changelog.md#0230)
This renders everything related to `stats` in the helm-chart obsolete. It is very confusing for the user to have a section in the `values.yaml` which does not do anything.

**Describe the solution you'd like**
Remove everything related to the now-removed stats-server from the chart.

**Describe alternatives you've considered**
We could also include the snippet from the [changelog](https://github.com/kubernetes/ingress-nginx/blob/master/Changelog.md#0230) to re-enable the http server but I think this is very advanced and should not be part of this chart.

### PR DESCRIPTION
**Pull request by fbreckle**
The stats nginx-server was removed from nginx-ingress, so references to
to its configration can be removed from the chart.

Signed-off-by: Fabian Breckle <fabian.breckle@breuninger.de>

@ChiefAlexander 
@taharah 

#### Which issue this PR fixes
  - fixes #16137

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/nginx-ingress/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: nginx-ingress
-version: 1.12.1
+version: 1.12.2
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/Chart.yaml]: Bump the minor version here. Another change is going in before yours so this will be `1.14.0`

**Review by ChiefAlexander — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you also add a small blob in the readme about how to enable the stats page using http-snipet.

File: stable/nginx-ingress/Chart.yaml (hunk 1 above)
Comment (fbreckle) [reply stable/nginx-ingress/Chart.yaml]: > Thanks for the PR! Could you also add a small blob in the readme about how to enable the stats page using http-snipet.

Does this make sense? For reference, the snipped is
```
http-snippet: |
  server {
    listen 18080;

    location /nginx_status {
      allow 127.0.0.1;
      allow ::1;
      deny all;
      stub_status on;
    }

    location / {
      return 404;
    }
  }
```

This means the webserver is only accessible from localhost. We could change the allowed IPs of course, but then we would want it configurable again. IMHO that feature is not worth the increased complexity of the chart. I also dont think that feature is meant to be used outside of local debugging.

Comment (ChiefAlexander) [reply stable/nginx-ingress/Chart.yaml]: We discussed on slack on how best to describe the change within the readme. What we settled on was to describe what the stats page is now for those who come from outdated links.

**Comment by ChiefAlexander** [comment]
Alright to get this in we need to resolve the git conflicts and you will need to follow https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work to sign your commits. After that this looks good!

**Comment by ChiefAlexander** [comment]
Tests are failing due to this test which is unrelated to your change -> https://github.com/helm/charts/blob/master/stable/nginx-ingress/ci/deamonset-hostport-values.yaml 
We have had problems with running this test a few times now, due to it requiring ports that can conflict with other pods. Go ahead and remove this test. In the future we will re-add it if we can get it to not be as flaky.

**Comment by ChiefAlexander** [comment]
/lgtm

Thank you very much!

**Pull request merged**